specs/waku/v2/waku-store.md

```diff
@@ -2,36 +2,88 @@
 title: Waku
 version: 2.0.0-beta1
 status: Draft
-authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
+authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>, Sanaz Taheri <sanaz@status.im>
 ---
 
 # Table of Contents
 
 - [Abstract](#abstract)
+  * [Waku Store Protocol](#Waku-Store-Protocol)
+  * [Pagination](#Pagination)
 - [Wire Specification](#wire-specification)
   * [Protobuf](#protobuf)
 - [Changelog](#changelog)
 - [Copyright](#copyright)
 
 # Abstract
 
+## Waku Store Protocol 
 `WakuStore` is a protocol to enable querying of messages received through relay protocol and stored by other nodes.
 
+## Pagination
+Nodes supporting store protocol MAY support Pagination. When a node supports pagination, a requester node can set the number of waku messages to return and the offset to start from. The queried node MUST then use this information to return messages in the queried range. The details of the pagination are explained next.
+
+To perform pagination, each `WakuMessage` stored at a waku store node is associated with a unique `Index`.  Once a `WakuMessage` is received by a node, it will generate the `Index` of the  `WakuMessage` and then saves the (`Index`, `WakuMessage`) together in its local storage. An `Index` encapsulates fields useful for paging, these fields are a `digest` and a `receivedTime`. The `receivedTime` indicates the time at which the `WakuMessage` is received by the store node and the `digest` is the hash of the received `WakuMessage`.  
```

Comment:
```suggestion
To perform pagination, each `WakuMessage` stored at a Waku node running the store protocol is associated with a unique `Index`. Once a `WakuMessage` is received by a node, it will generate the `Index` of the  `WakuMessage` and then saves the (`Index`, `WakuMessage`) together in its local storage. An `Index` encapsulates fields useful for paging, these fields are a `digest` and a `receivedTime`. The `receivedTime` indicates the time at which the `WakuMessage` is received by the store node and the `digest` is the hash of the received `WakuMessage`.  
```